libs/ssh/keys.go

```diff
@@ -127,3 +128,41 @@ func checkAndGenerateSSHKeyPair(ctx context.Context, keyPath string) (string, st
 
 	return keyPath, strings.TrimSpace(string(publicKeyBytes)), nil
 }
+
+func checkAndGenerateSSHKeyPairFromSecrets(ctx context.Context, client *databricks.WorkspaceClient, clusterID, privateKeyName, publicKeyName string) ([]byte, []byte, error) {
+	secretsScopeName, err := createSecretsScope(ctx, client, clusterID)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to create secrets scope: %w", err)
+	}
+
+	privateKeyBytes, err := getSecret(ctx, client, secretsScopeName, privateKeyName)
+	if err != nil {
+		cmdio.LogString(ctx, "SSH key pair not found in secrets scope, generating a new one...")
+
+		privateKeyBytes, publicKeyBytes, err := generateSSHKeyPair()
+		if err != nil {
+			return nil, nil, fmt.Errorf("failed to generate SSH key pair: %w", err)
+		}
+
+		err = putSecret(ctx, client, secretsScopeName, privateKeyName, string(privateKeyBytes))
+		if err != nil {
+			return nil, nil, err
+		}
+
+		err = putSecret(ctx, client, secretsScopeName, publicKeyName, string(publicKeyBytes))
+		if err != nil {
+			return nil, nil, err
+		}
+
+		return privateKeyBytes, publicKeyBytes, nil
+	} else {
+		cmdio.LogString(ctx, "Using SSH key pair from secrets scope")
```

Comment:
Is this relevant to print when it is nominal operations?

You can consider using `libs/log` to emit a debug log message instead. Those are shown in a friendly format when the CLI is run with `--debug` or `--log-level debug`. You can use the trace level for higher verbosity.